web_api.md

```diff
@@ -0,0 +1,318 @@
+    password: "1234"
+hosts:
+  - address: localhost
+    username: cloud
+    password: "1234"
+    deploy_processors:
+      - name: ocrd-cis-ocropy-binarize
+        number_of_instance: 2
+        deploy_type: native
+      - name: ocrd-olena-binarize
+        number_of_instance: 1
+        deploy_type: docker
+
+  - address: 134.76.1.1
+    username: tdoan
+    path_to_privkey: /path/to/file
+    deploy_processors:
+      - name: ocrd-eynollah-segment
+        number_of_instance: 1
+        deploy_type: native
+```
+
+There are three main sections in the configuration file.
+
+1. `message_queue`: it contains the `address` and `port`, where the queue was deployed, or will be deployed with the
+   specified `credentials`. If the `ssh` property is presented, the Processing Server will try to connect to
+   the `address` via `ssh` with provided `username` and `password` and deploy [RabbitMQ](https://www.rabbitmq.com/) at
+   the specified `port`. The remote machine must have [Docker](https://www.docker.com/) installed since the deployment
+   is done via Docker. Make sure that the provided `username` has enough rights to run Docker commands. In case
+   the `ssh` property is not presented, the Processing Server assumes that RabbitMQ was already deployed and just uses
+   it.
+2. `mongo_db`: this section also contains the `address` and `port`, where the [MongoDB](https://www.mongodb.com/) is
+   running, or will run. If `credentials` is presented, it will be used when deploying and connecting to the database.
+   The `ssh` section behaves exactly the same as described in the `message_queue` section above.
+3. `hosts`: this section contains a list of hosts, usually virtual machines, where Processing Workers should be
+   deployed. To be able to connect to a host, an `address` and `username` are required, then comes either `password`
+   or `path_to_privkey` (path to a private key). All Processing Workers, which will be deployed, must be declared under
+   the `deploy_processors` property. In case `type` is `docker`, make sure that [Docker](https://www.docker.com/) is
+   installed in the target machine and the provided `username` has enough rights to execute Docker commands.
+
+Among three sections, only the `message_queue` is required. However, if `hosts` is present, `mongo_db` must be there
+as well. For more information, please check the [configuration file schema](web_api/config.schema.yml).
+
+### Processing Worker
+
+There is normally no need to start (or stop) a Processing Worker manually, since it can be managed by a Processing
+Server via a [configuration file](#processing-server). However, if it is necessary to do so, there are two ways to start
+a Processing Worker:
+
+```shell
+# 1. Use ocrd CLI bundled with OCR-D/core
+$ ocrd processing-worker <processor-name> --queue=<queue-address> --database=<database-address>
+
+# 2. Use processor name
+<processor-name> processing-worker --queue=<queue-address> --database=<database-address>
```

Comment:
Are you sure about this mnemonic? It strikes me as inconsistent with the existing CLI conventions.

For the [REST wrapper](https://github.com/OCR-D/core/pull/884), you have added `--address`. Is this whole functionality obsolete now? (I thought that the reference implementation could easily provide both a worker for the queue model and a standalone REST wrapper along #884's lines, i.e. `POST /` etc. So maybe for the CLI something like `--server=worker --queue=... --database=...` and `--server=rest --address=...` could be defined.)

Reply:
> Is this whole functionality obsolete now?

For my initial idea, yes. I wanted to support only the queue architecture, but after reading your comments, I think it's better to have both, queue and standalone REST wrapper. The CLI syntax you suggested is a good idea.

Reply:
How about a syntax like this?

```shell
# Start a worker
$ ocrd server --type=worker --queue=... --database=...
$ ocrd-dummy server --type=worker --queue=... --database=...

# Standalone server
$ ocrd server --type=rest --address=... --database=...
$ ocrd-dummy server --type=rest --queue=... --database=...
```

Reply:
Yes, I guess your cmd-group syntax is clearer. And it does help keep the standard help text short.

Reply:
Also, subcommands for the processor CLI are really tricky to do, so if it was to be based on the `ocrd-foo-processor` CLI, the worker arguments would realistically need to be options of the `ocrd-foo-processor` CLI, which makes the help text longer and requires more if-else-clauses to detect starting a worker vs. normal processing vs. `--help` etc.

So, I'm :+1: for a subcommand to `ocrd`.

Reply:
@kba perhaps you can comment on the respective [discussion in the Web API](https://github.com/OCR-D/spec/pull/222#issuecomment-1425934842) then.